RoddyCore/src/de/dkfz/roddy/plugins/LibrariesFactory.groovy

```diff
@@ -531,10 +559,12 @@ public class LibrariesFactory extends Initializable {
             // version of it which may either be a revision (x:x.y-[0..n] or a higher compatible version.
             // Search the last valid entry in the chain.
             if (!usedPlugins.contains("${id}:${version}")) {
-                for (; pInfo.nextInChain != null; pInfo = pInfo.nextInChain) {
+                while (true) {
                     version = pInfo.prodVersion;
                     if (usedPlugins.contains("${id}:${version}")) //Break, if the list of used plugins contains the selected version of the plugin
-                        break;
+                        break
+                    if (pInfo.nextInChain == null) break // Break if this was the last entry in the chain
```

Comment:
The previous version with the for-loop was not nice, but still I think it was more readable.